ivy/functional/frontends/tensorflow/math.py

```diff
@@ -297,3 +297,23 @@ def zero_fraction(value, name="zero_fraction"):
 
 
 # TODO: Ibeta for Future Release
+
+
+def accumulate_n(inputs, shape=None, tensor_dtype=None, name="accumulate_n"):
+    if tensor_dtype is None:
+        tensor_dtype = ivy.dtype(inputs[0])
+    if shape is None:
+        shape = ivy.shape(inputs[0])
+    # shape and dtype check
+    for i in range(len(inputs)):
+        ivy.assertions.check_equal(
```

Comment:
According to the definition for `ivy.assertions.check_equal` is only compares `x1` and `x2`. I don't think this will work, you would need to iterate and compare input[n] to input[n+1] for all inputs. Then AND all the return booleans. Or maybe there is a better solution to this!